src/WebJobs.Script/Description/DotNet/FSharp/FSharpCompilationService.cs

```diff
@@ -123,31 +114,16 @@ public ICompilation GetFunctionCompilation(FunctionMetadata functionMetadata)
                 // as dictated by the C# reference resolver, and F# doesn't like getting multiple references to those.
                 otherFlags.Add("--noframework");
 
-                // Add the references as reported by the C# reference resolver.
-                foreach (var mdr in compilation.References)
-                {
-                    if (!mdr.Display.Contains("Unresolved "))
+                var references = script.Options.MetadataReferences
+                    .Cast<UnresolvedMetadataReference>()
+                    .Aggregate(new List<PortableExecutableReference>(), (a, r) =>
                     {
-                        otherFlags.Add("-r:" + mdr.Display);
-                    }
-                }
+                        a.AddRange(_metadataResolver.ResolveReference(r.Reference, string.Empty, r.Properties));
+                        return a;
+                    });
 
-                // Above we have used the C# reference resolver to get the basic set of DLL references for the compilation.
-                //
-                // However F# has its own view on default options. For scripts these should include the
-                // following framework facade references.
-
-                otherFlags.Add("-r:System.Linq.dll"); // System.Linq.Expressions.Expression<T> 
```

Comment:
Btw, I did try using types that (as defined here) would depend on those references and didn't seem to run into issues. I think it would probably be a good idea to have some tests covering those for validation.